docs/src/main/sphinx/sql/set-session.rst

```diff
@@ -15,28 +15,41 @@ Description
 
 Set a session property value or a catalog session property.
 
+.. _session-properties-definition:
 
-Connectors can provide session properties. They are set separately for each
-catalog by prefixing them with the catalog name. There is no sharing of
-properties across catalogs, even if the catalogs use the same connector.
+Session properties
+------------------
+
+A session property is a :doc:`configuration property </admin/properties>` that
+can be temporarily modified by a user for the duration of their current
+connection to the Trino cluster. Many configuration properties have a
+corresponding session property that accepts the same values as the config
+property.
+
+A catalog session property is a connector-defined session property that is set
+on a per-catalog basis. These properties are set separately for each catalog
+by including the catalog name as a prefix.
+
+If the user disconnects or establishes another connection to the Trino cluster,
+the values of all session properties and catalog session properties are restored
+to the cluster and its catalog's existing configuration.
```

Comment:
Misleading. It's sufficient to mention that "temporarily modified by a user for the duration of their current
connection to the Trino cluster".

Nothing is there to reset or restore.

If the intent is to emphasize that session properties are lost once session ends then it could be worded like:

```
Session property changes are tied to the current session and once the session ends (either via disconnecting or creation a new connection) any changes made are lost.

This also means that you have can have two different connections with both setting the same session property to different values.
```